### DISCUSSION TITLE
test notification arrives but not the purchase ones

### DISCUSSION THREAD

**Discussion by henry-ht** [imdhemy/laravel-in-app-purchases#137] [Q&A]
Hello, sorry for the question. I'll explain my situation: when I send the test notification it reaches the server and I see it in the log, but when I make the test purchase the notification never reaches my server.

**Comment by imdhemy** [comment]
Could you please provide the following info?

- Which LIAP version?
- Google or Apple?
- Did you check the Requests logs?
- Did updated the config file by adding the listeners? 

---
_I will be away from keyboard for 10 days, so I'm following up with my mobile phone._

**Reply by henry-ht** [reply]
Thank you.

I have version 1.0 of liap, I'm going to update to the new one.

I am using google notifications.

If I check the log at all times and I only see when I send the test notification from the google play console

and my listeners

![image](https://user-images.githubusercontent.com/15837863/189171001-e7249bc8-758c-45bf-bf80-0efb0548ade7.png)

**Reply by imdhemy** [reply]
I suggest you to use ngrok to expose your local environment. Update Google settings to receive notifications on local host. Try to dump and die the request at this line.

https://github.com/imdhemy/laravel-in-app-purchases/blob/22d905f8bfb9eafdc3680b8018c1fca718995830/src/Http/Controllers/ServerNotificationController.php#L27

Try to debug this way in different places to find the cause of the problem, if it’s related to liap, I will publish a fix asap

**Reply by henry-ht** [reply]
Thank you.

**Reply by henry-ht** [reply]
hello again. can you help me with this i get a 403 error - unauthorized

**Reply by henry-ht** [reply]
I have this in the .env file "GOOGLE_APPLICATION_CREDENTIALS" but I don't think the app is receiving it... can you show me an example of how to add it to the root of the project and call this file?

**Reply by imdhemy** [reply]
For the credentials path just use the absolute path for example

'/var/www/html/path/to/creds.json'

---

But I guess you are using signed routes, the signature depends on the application secret key. Changing the secret key used while generating the signed routes, will result in unauthorised response.

**Comment by henry-ht** [comment]
I already validated everything, the notification from google rdtn arrives in the library and gives an unauthorized 403 error

**Comment by henry-ht** [comment]
I'm checking if I should add my listeners in another part of the laravel code

**Comment by henry-ht** [comment]
the library recovers the data perfectly from google play but for some reason, which I do not know. gives a 403 error as a response.

I hope your help

**Reply by henry-ht** [reply]
`[2022-09-08 21:10:45] local.ERROR: include(E:\localhost\myMiniStore\vendor\composer/../imdhemy/laravel-purchases/src/ServiceProviders/LiapServiceProvider.php): Failed to open stream: No such file or directory {"exception":"[object] (ErrorException(code: 0): include(E:\\localhost\\myMiniStore\\vendor\\composer/../imdhemy/laravel-purchases/src/ServiceProviders/LiapServiceProvider.php): Failed to open stream: No such file or directory at E:\\localhost\\myMiniStore\\vendor\\composer\\ClassLoader.php:571)
[stacktrace]
#0 E:\\localhost\\myMiniStore\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Bootstrap\\HandleExceptions.php(255): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleError(2, 'include(E:\\\\loca...', 'E:\\\\localhost\\\\my...', 571)
#1 E:\\localhost\\myMiniStore\\vendor\\composer\\ClassLoader.php(571): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->Illuminate\\Foundation\\Bootstrap\\{closure}(2, 'include(E:\\\\loca...', 'E:\\\\localhost\\\\my...', 571)
#2 E:\\localhost\\myMiniStore\\vendor\\composer\\ClassLoader.php(571): include('E:\\\\localhost\\\\my...')
#3 E:\\localhost\\myMiniStore\\vendor\\composer\\ClassLoader.php(428): Composer\\Autoload\\includeFile('E:\\\\localhost\\\\my...')
#4 E:\\localhost\\myMiniStore\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Application.php(749): Composer\\Autoload\\ClassLoader->loadClass('Imdhemy\\\\Purchas...')
#5 E:\\localhost\\myMiniStore\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Application.php(683): Illuminate\\Foundation\\Application->resolveProvider('\\\\Imdhemy\\\\Purcha...')
#6 E:\\localhost\\myMiniStore\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\ProviderRepository.php(75): Illuminate\\Foundation\\Application->register('\\\\Imdhemy\\\\Purcha...')
#7 E:\\localhost\\myMiniStore\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Application.php(663): Illuminate\\Foundation\\ProviderRepository->load(Array)
#8 E:\\localhost\\myMiniStore\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Bootstrap\\RegisterProviders.php(17): Illuminate\\Foundation\\Application->registerConfiguredProviders()
#9 E:\\localhost\\myMiniStore\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Application.php(239): Illuminate\\Foundation\\Bootstrap\\RegisterProviders->bootstrap(Object(Illuminate\\Foundation\\Application))
#10 E:\\localhost\\myMiniStore\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Console\\Kernel.php(310): Illuminate\\Foundation\\Application->bootstrapWith(Array)
#11 E:\\localhost\\myMiniStore\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Console\\Kernel.php(127): Illuminate\\Foundation\\Console\\Kernel->bootstrap()
#12 Command line code(1): Illuminate\\Foundation\\Console\\Kernel->handle(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))`

**Reply by imdhemy** [reply]
Could you please submit a PR with the following fix:

Remove the two back slashes in the beginning of this line.

https://github.com/imdhemy/laravel-in-app-purchases/blob/22d905f8bfb9eafdc3680b8018c1fca718995830/composer.json#L53

**Reply by henry-ht** [reply]
I apologize, I don't speak the language and I'm somewhat undecided on how to do the pr here in githud.

but if you guide me I could help.

**Reply by imdhemy** [reply]
In this repository click the (fork) button. A copy of this repository will be created on your profile. Edit the file and save it. A new blue button will appear on GitHub that will help you to create a pull request.

**Reply by imdhemy** [reply]
Thanks to @iabdullahbeker 
I just published a fix.

https://github.com/imdhemy/laravel-in-app-purchases/releases/tag/1.1.1

Could you please try again.
